flink-core/src/test/java/org/apache/flink/api/common/io/BinaryInputFormatTest.java

```diff
@@ -160,11 +161,45 @@ public void testGetStatisticsMultiplePaths() throws IOException {
                 stats.getTotalInputSize());
     }
 
+    @Test
+    public void testCreateInputSplitsWithEmptySplit() throws IOException {
+        final int blockInfoSize = new BlockInfo().getInfoSize();
+        final int blockSize = blockInfoSize + 8;
+        final int numBlocks = 3;
+        final int minNumSplits = 5;
+
+        // create temporary file with 3 blocks
+        final File tempFile =
+                createBinaryInputFile(
+                        "test_create_input_splits_with_empty_split", blockSize, numBlocks);
+
+        final Configuration config = new Configuration();
+        config.setLong("input.block_size", blockSize + 10);
+
+        final BinaryInputFormat<Record> inputFormat = new MyBinaryInputFormat();
+        inputFormat.setFilePath(tempFile.toURI().toString());
+        inputFormat.setBlockSize(blockSize);
+
+        inputFormat.configure(config);
+
+        FileInputSplit[] inputSplits = inputFormat.createInputSplits(minNumSplits);
+
+        Assert.assertEquals(
```

Comment:
Are you willing to use assertj instead of junit.Assert,so that subsequent migration to junit5 will be more convenient.

Reply:
Makes sense. Ref: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

Reply:
Maybe add a hotfix commit "Rework BinaryInputFormatTest to be based on AssertJ" first. And then add the new test using AssertJ.

Reply:
Thanks for reviewing, the suggestion is very helpful. I have reworked BinaryInputFormatTest to be based on AssertJ, can you take a look when you are free？